src/LabelSelector/LabelSelector.js

```diff
@@ -314,6 +315,12 @@ LabelSelector.propTypes = {
    * @see https://material-ui.com/components/autocomplete/#limit-tags
    */
   limitTags: PropTypes.number,
+  /**
+   * The maximum length of a label name.
+   * @default 50
+   * @type {number}
+   */
+  maxLabelLength: PropTypes.number,
```

Comment:
Does this prop just relate to name or description too? I think the prop name `maxLabelLength` is ambigious. If it applies to name, which I suspect it does, then maybe `nameMaxLength` might be clearer and future proofs if we ever want to limit description field too.